### PR TITLE
adds base support for mockery tool in Makefile

### PR DESCRIPTION
**Pull request by vjayaramrh**

### CODE REVIEW

File: default-mockery.mk [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+include $(GIT_ROOT_DIR)/detect-container-runtime.mk
+
+
+.PHONY: install-mockery
+install-mockery: ## install mockery
+ifeq ($(CONTAINER_RUNNABLE), 0)
+		$(CONTAINER_RUNTIME) pull docker.io/vektra/mockery:v${MOCKERY_VERSION}
+else
+		wget -qO- https://github.com/vektra/mockery/releases/download/v${MOCKERY_VERSION}/mockery_${MOCKERY_VERSION}_${OS}_${OS_ARCH}.tar.gz | sudo tar -xvzf - -C /usr/local/bin
+endif
```
Comment (dkosteck) [inline default-mockery.mk:29]:
```suggestion
.PHONY: generate-mocks
generate-mocks:
ifeq ($(CONTAINER_RUNNABLE), 0)
		$(CONTAINER_RUNTIME) run --security-opt label=disable -v ${PWD}:/src -w /src docker.io/vektra/mockery:v${MOCKERY_VERSION}
else
		mockery
endif
```

Comment (dkosteck) [reply default-mockery.mk:29]: Add the generate mocks target from https://github.com/nephio-project/nephio/pull/441

**Review by efiacor — COMMENTED** [review]
I know I am using it incorrectly but please advise. Thanks.

`{mockery_makefile} nephio$ gs
On branch mockery_makefile
Untracked files:
  (use "git add <file>..." to include in what will be committed)
        .mockery.yaml

nothing added to commit but untracked files present (use "git add" to track)
 {mockery_makefile} nephio$ gd
 {mockery_makefile} nephio$ cat .mockery.yaml 
packages:
  github.com/nephio-project/nephio/controllers/pkg/giteaclient:
    interfaces:
      GiteaClient:
        config:
          dir: "{{.InterfaceDir}}" {mockery_makefile} nephio$ make generate-mocks^C
 {mockery_makefile} nephio$ make generate-mocks
docker run --security-opt label=disable -v /home/fiachra/git/nephio/nephio:/src -w /src docker.io/vektra/mockery:v2.37.1
07 Dec 23 09:42 UTC INF Starting mockery dry-run=false version=v2.37.1
07 Dec 23 09:42 UTC INF Using config: /src/.mockery.yaml dry-run=false version=v2.37.1
error occurred when loading packages
07 Dec 23 09:42 UTC ERR encountered error when loading package error="-: no required module provides package github.com/nephio-project/nephio/controllers/pkg/giteaclient: go.mod file not found in current directory or any parent directory; see 'go help modules'" dry-run=false version=v2.37.1
07 Dec 23 09:42 UTC ERR unable to parse packages error="error occurred when loading packages" dry-run=false version=v2.37.1
make: *** [default-mockery.mk:34: generate-mocks] Error 1
 {mockery_makefile} nephio$ cd controllers/pkg/
 {mockery_makefile} pkg$ make generate-mocks
make: *** No rule to make target 'generate-mocks'.  Stop.`

**Comment by vjayaramrh** [comment]
> I know I am using it incorrectly but please advise. Thanks.
> 
> ```shell
> {mockery_makefile} nephio$ gs
> On branch mockery_makefile
> Untracked files:
>   (use "git add <file>..." to include in what will be committed)
>         .mockery.yaml
> 
> nothing added to commit but untracked files present (use "git add" to track)
> 
>  {mockery_makefile} nephio$ cat .mockery.yaml 
> packages:
>   github.com/nephio-project/nephio/controllers/pkg/giteaclient:
>     interfaces:
>       GiteaClient:
>         config:
>           dir: "{{.InterfaceDir}}" 
> 
>  {mockery_makefile} nephio$ make generate-mocks
> docker run --security-opt label=disable -v /home/fiachra/git/nephio/nephio:/src -w /src docker.io/vektra/mockery:v2.37.1
> 07 Dec 23 09:42 UTC INF Starting mockery dry-run=false version=v2.37.1
> 07 Dec 23 09:42 UTC INF Using config: /src/.mockery.yaml dry-run=false version=v2.37.1
> error occurred when loading packages
> 07 Dec 23 09:42 UTC ERR encountered error when loading package error="-: no required module provides package github.com/nephio-project/nephio/controllers/pkg/giteaclient: go.mod file not found in current directory or any parent directory; see 'go help modules'" dry-run=false version=v2.37.1
> 07 Dec 23 09:42 UTC ERR unable to parse packages error="error occurred when loading packages" dry-run=false version=v2.37.1
> make: *** [default-mockery.mk:34: generate-mocks] Error 1
> 
>  {mockery_makefile} nephio$ cd controllers/pkg/
> 
>  {mockery_makefile} pkg$ make generate-mocks
> make: *** No rule to make target 'generate-mocks'.  Stop.
> ```

@efiacor This PR does not have any `.mockery.yaml` file, hence `make generate-mocks` is erroring out.

**Review by efiacor — APPROVED** [review]

**Review by liamfallon — COMMENTED** [review]
/approve

**Comment by liamfallon** [comment]
I have pulled this PR and it works well. I have added tests in the code based on it.

**Pull request merged**